include/boost/windows_sspi/stream.hpp

```diff
@@ -319,6 +365,24 @@ class stream : public stream_base {
         async_write_impl<next_layer_type, ConstBufferSequence>{m_next_layer, buffer, m_sspi_impl}, token);
   }
 
+  void shutdown(boost::system::error_code& ec) {
+    switch(m_sspi_impl.shutdown()) {
+      case detail::sspi_shutdown::state::data_available:
+        net::write(m_next_layer, net::buffer(m_sspi_impl.shutdown.data()), net::transfer_exactly(m_sspi_impl.shutdown.data().size()), ec);
+        return;
+      case detail::sspi_shutdown::state::error:
+        ec = m_sspi_impl.shutdown.last_error();
+    }
+  }
+
+  template <typename CompletionToken>
+  auto async_shutdown(CompletionToken&& token) ->
+    typename net::async_result<typename std::decay<CompletionToken>::type,
+                               void(boost::system::error_code)>::return_type {
+    return boost::asio::async_compose<CompletionToken, void(boost::system::error_code)>(
```

Comment:
Shouldn't this be `net::async_compose`?

Reply:
I'm fairly certain `async_compose` is only available in `boost::asio` whereas the `net` namespace alias is used for things that are available in the networking TS.

This was written with the help of @vinniefalco who specifically pointed this out.